_posts/2018-11-12-writing-fast-rails-part-2.markdown

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title:  "Tips for Writing Fast Rails: Part 2"
+date: 2018-11-12 12:10:00
+categories: ["rails", "performance"]
+author: "luciano"
+---
+
+Some time ago we wrote a couple of [Tips for Writing Fast Rails](https://www.ombulabs.com/blog/performance/rails/writing-fast-rails.html). Now we decided to continue sharing our knowledge on this topic so here is the part two!
+
+<!--more-->
+
+A common mistake that we usually see in [Rails](https://rubyonrails.org/) applications is the lack of use of [ActiveRecord::Calculations](https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html), which is the rails way to do math operations in the database. This can affect the performance of your application because otherwise you'll have to do it in the code. That means to initialize [ActiveRecord](https://guides.rubyonrails.org/active_record_basics.html) objects and iterate through them, which is not the faster way to do it, specially when you have a ton of records in your database.
```

Comment:
@lubc I think it would be a good idea to use a title per tip. Just like we did in the previous article:

> Prefer where instead of select

This section could say something like: 

> Prefer `ActiveRecord::Calculations` instead of `Array` calculations

Also, this could be a little more straightforward: 

> A common mistake that we usually see in [Rails](https://rubyonrails.org/) applications is the lack of use of [ActiveRecord::Calculations](https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html), which is the rails way to do math operations in the database. 

Like this: 

> Usually in [Rails](https://rubyonrails.org/) applications we find many references to `Array::sum` for summing values. This is a common mistake because [ActiveRecord::Calculations](https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html) provides a way to do this without loading a bunch of `ActiveRecord` objects in memory. If you want to perform mathematical operations for a set of records following the Rails way, `ActiveRecord::Calculations` is the best way to do them in the database.